src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1428,6 +1466,35 @@ impl Decimal {
         Res::Ok(x)
     }
 
+    /// `int_part` returns int part of the decimal. It's temporary and
+    /// after we adjust `as_i64`, it will be removed.
+    pub fn int_part(&self, ctx: &EvalContext) -> ::std::result::Result<i64, ExprError> {
```

Comment:
I think you can write a function to deal with both `Res` and truncate strategy instead.